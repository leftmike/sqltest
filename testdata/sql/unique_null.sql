--
-- Test UNIQUE constraints with NULLs
--

DROP TABLE IF EXISTS tbl1;

CREATE TABLE tbl1 (
    c1 int primary key,
    c2 int unique,
    c3 int
);

INSERT INTO tbl1 VALUES
    (0, NULL, 0),
    (1, NULL, 0),
    (2, NULL, 0),
    (3, 40, 0);

SELECT * FROM tbl1;

{{Fail .Test}}
INSERT INTO tbl1 VALUES
    (4, 40, 0);

{{Fail .Test}}
UPDATE tbl1 SET c2 = 40 WHERE c1 = 2;

UPDATE tbl1 SET c2 = 30 WHERE c1 = 2;

SELECT * FROM tbl1;

UPDATE tbl1 SET c2 = NULL WHERE c1 = 3;

UPDATE tbl1 SET c2 = 40 WHERE c1 = 2;

SELECT * FROM tbl1;

{{Fail .Test}}
UPDATE tbl1 SET c2 = 40 WHERE c1 = 1;

INSERT INTO tbl1 VALUES
    (4, 60, 0),
    (5, NULL, 0),
    (6, 80, 0),
    (7, NULL, 0);

SELECT * FROM tbl1;
