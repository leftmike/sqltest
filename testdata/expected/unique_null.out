--
-- Test UNIQUE constraints with NULLs
--
DROP TABLE IF EXISTS tbl1;
CREATE TABLE tbl1 (
    c1 int primary key,
    c2 int unique,
    c3 int
);
INSERT INTO tbl1 VALUES
    (0, NULL, 0),
    (1, NULL, 0),
    (2, NULL, 0),
    (3, 40, 0);
SELECT * FROM tbl1;
   c1 c2 c3
   -- -- --
 1  0     0
 2  1     0
 3  2     0
 4  3 40  0
(4 rows)
{{Fail .Test}}
INSERT INTO tbl1 VALUES
    (4, 40, 0);
{{Fail .Test}}
UPDATE tbl1 SET c2 = 40 WHERE c1 = 2;
UPDATE tbl1 SET c2 = 30 WHERE c1 = 2;
SELECT * FROM tbl1;
   c1 c2 c3
   -- -- --
 1  0     0
 2  1     0
 3  2 30  0
 4  3 40  0
(4 rows)
UPDATE tbl1 SET c2 = NULL WHERE c1 = 3;
UPDATE tbl1 SET c2 = 40 WHERE c1 = 2;
SELECT * FROM tbl1;
   c1 c2 c3
   -- -- --
 1  0     0
 2  1     0
 3  2 40  0
 4  3     0
(4 rows)
{{Fail .Test}}
UPDATE tbl1 SET c2 = 40 WHERE c1 = 1;
INSERT INTO tbl1 VALUES
    (4, 60, 0),
    (5, NULL, 0),
    (6, 80, 0),
    (7, NULL, 0);
SELECT * FROM tbl1;
   c1 c2 c3
   -- -- --
 1  0     0
 2  1     0
 3  2 40  0
 4  3     0
 5  4 60  0
 6  5     0
 7  6 80  0
 8  7     0
(8 rows)
